go/glide/lib.h

```diff
@@ -0,0 +1,88 @@
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+/**
+ * FFI compatible version of the ErrorType enum defined in protobuf.
+ */
+typedef enum ErrorType {
+  ClosingError = 0,
+  RequestError = 1,
+  TimeoutError = 2,
+  ExecAbortError = 3,
+  ConnectionError = 4,
+} ErrorType;
+
+/**
+ * A Redis error.
+ */
+typedef struct RedisErrorFFI {
+  const char *message;
+  enum ErrorType error_type;
+} RedisErrorFFI;
+
+/**
+ * The connection response.
+ *
+ * It contains either a connection or an error. It is represented as a struct instead of an enum for ease of use in the wrapper language.
+ */
+typedef struct ConnectionResponse {
+  const void *conn_ptr;
+  const struct RedisErrorFFI *error;
+} ConnectionResponse;
+
+/**
+ * Success callback that is called when a Redis command succeeds.
+ */
+typedef void (*SuccessCallback)(uintptr_t channel_address, const char *message);
+
+/**
+ * Failure callback that is called when a Redis command fails.
+ */
+typedef void (*FailureCallback)(uintptr_t channel_address, const struct RedisErrorFFI *error);
+
+/**
+ * Creates a new client to the given address. The success callback needs to copy the given string synchronously, since it will be dropped by Rust once the callback returns. All callbacks should be offloaded to separate threads in order not to exhaust the client's thread pool.
+ *
+ * # Safety
+ *
+ * * `connection_request_bytes` must point to `connection_request_len` consecutive properly initialized bytes.
+ * * `connection_request_len` must not be greater than `isize::MAX`. See the safety documentation of [`std::slice::from_raw_parts`](https://doc.rust-lang.org/std/slice/fn.from_raw_parts.html).
+ */
+const struct ConnectionResponse *create_client(const uint8_t *connection_request_bytes,
```

Comment:
Please, add a doc describing that `ConnectionResponse` should be freed afterwards regardless of the error

Reply:
Added.